client/src/Add.js

```diff
@@ -0,0 +1,83 @@
+import { useState } from "react";
+
+
+function Add({ data, setVideoData, setSorted })
+{
+    const [newTitle, setNewTitle] = useState("");
+    const [newURL, setNewURL] = useState("");
+
+    let newItem = {};
+
+
+    const addFunction = (data) => 
+    {
+        newItem.id = data.length;
+        newItem.title = newTitle;
+        newItem.url = newURL;
+        newItem.rating = 0;
+        newItem.added = new Date();
+
+        if (newItem.title !== "" && newItem.url !== "")
+        {
+            setVideoData(previous => [...previous, newItem]);
+            setSorted(false);
+        }
+    }
+
+    const handleTitle = (event) => 

```

Comment:
consider using a code formatter to get your indentations correct

Reply:
This is C# formatting style.